### PR TITLE
test first ver

### PR DESCRIPTION
**Pull request by LinkClink**

### CODE REVIEW

**Review by DzhuzhaMichael — APPROVED** [review]

**Comment by DzhuzhaMichael** [comment]
Looks good!

File: src/test/java/core/basesyntax/service/impl/FileWriteServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    @BeforeClass
+    public static void setUp() {
+        goodData = "fruit,quality" + System.lineSeparator()
+                + "banana,152" + System.lineSeparator()
+                + "apple,90";
+        goodFilePath = "src/test/resources/out.csv";
+    }
+
+    @Test
+    public void fileWrite_goodFilePath_Ok() throws IOException {
```
Comment (KostyaNikiforov) [inline src/test/java/core/basesyntax/service/impl/FileWriteServiceImplTest.java:26]:
```suggestion
    public void fileWrite_goodFilePath_Ok() {
```
it is not necessary

File: src/test/java/core/basesyntax/service/impl/GenerateReportDataImplTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void generateReport_emptyData_Ok() {
+        String actual = generateReportDataService.report(data);
+        String expected = "fruit,quality"
+                + System.lineSeparator();
+        assertEquals(expected, actual);
+    }
```
Comment (KostyaNikiforov) [inline src/test/java/core/basesyntax/service/impl/GenerateReportDataImplTest.java:32]: It would be cool to add here at least one negative test case

File: src/test/java/core/basesyntax/service/impl/WriteDataToStorageImplTest.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import core.basesyntax.service.file.WriteDataToStorageService;
+import org.junit.After;
+import org.junit.Test;
+
+public class WriteDataToStorageImplTest {
+    private WriteDataToStorageService writeDataToStorageService = new WriteDataToStorageImpl();
+    private TransactionStorage transactionStorage = new TransactionStorage();
+
+    @Test
+    public void writeDataToStorage_firstLine_Ok() {
```
Comment (KostyaNikiforov) [inline src/test/java/core/basesyntax/service/impl/WriteDataToStorageImplTest.java:15]: First part of test method name should be name of the method you test 
```suggestion
    public void writeData_firstLine_Ok() {
```

**Review by KostyaNikiforov — CHANGES_REQUESTED** [review]
Let's add everywhere a test case when the input is null

**Review by KostyaNikiforov — APPROVED** [review]

File: src/main/java/core/basesyntax/service/impl/FileWriteServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+public class FileWriteServiceImpl implements FileWriteService {
+
+    @Override
+    public void writeDataToFile(String data, String filePath) {
+        try (BufferedWriter bufferedWriter
+                     = new BufferedWriter(new FileWriter(filePath, false))) {
+            bufferedWriter.write(data);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write to file" + e);
```
Comment (liliia-ponomarenko) [inline src/main/java/core/basesyntax/service/impl/FileWriteServiceImpl.java:16]:
```suggestion
            throw new RuntimeException("Can't write data to file" + e);
```
